### ISSUE TITLE
ITKR Installation Issues After Updating R

### ISSUE THREAD

**Issue by AlexandraBatzdorf** [stnava/ITKR#57]
Hello! I was able to install and run ITKR using R version 4.0, but in both 4.1.3 and 4.2.0, I can get about 82-84% of the way through before installation fails. It's compiled enough that I can call it using library(ITKR), but when I do, it gives the following error:

> Error: package or namespace load failed for ‘ITKR’ in namespaceExport(ns, exports):
>  undefined exports: itkCompileFlags, itkDir, itkIncludes, itkLibs, itkTag, itkVersion

This is the session info:

> R version 4.2.0 (2022-04-22)
> Platform: aarch64-apple-darwin20 (64-bit)
> Running under: macOS Monterey 12.1
> 
> Matrix products: default
> LAPACK: /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRlapack.dylib

this is how far the installation gets:

> [ 82%] Building CXX object Modules/ThirdParty/GDCM/src/gdcm/Source/MediaStorageAndFileFormat/CMakeFiles/gdcmMSFF.dir/__/__/Utilities/gdcmrle/info.cxx.o
> [ 82%] Building CXX object Modules/ThirdParty/GDCM/src/gdcm/Source/MediaStorageAndFileFormat/CMakeFiles/gdcmMSFF.dir/__/__/Utilities/gdcmrle/io.cxx.o
> [ 82%] Linking CXX static library ../../../../../../../lib/libitkgdcmMSFF-5.2.a
> [ 82%] Built target gdcmMSFF
> make[1]: *** [all] Error 2
> make: *** [libs] Error 2
> ERROR: compilation failed for package ‘ITKR’
> * removing ‘/Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/library/ITKR’
> Warning message:
> In i.p(...) :
>   installation of package ‘/var/folders/py/nyy7zwy505s32zc86rjh9vz80000gn/T//RtmpsTHeX0/filec9415c747238/ITKR_0.5.3.3.tar.gz’ had non-zero exit status

and here are the other warnings generated:

> [ 76%] Building CXX object Modules/Core/Common/src/CMakeFiles/ITKCommon.dir/itkMultiThreaderBase.cxx.o
> /private/var/folders/py/nyy7zwy505s32zc86rjh9vz80000gn/T/RtmpyN0ZpC/R.INSTALLc9b11febc519/ITKR/src/itks/Modules/Core/Common/src/itkMultiThreaderBase.cxx:545:17: warning: variable 'pixelCount' set but not used [-Wunused-but-set-variable]
>   SizeValueType pixelCount = 1;
>                 ^

> [ 80%] Building CXX object Modules/Core/Common/src/CMakeFiles/ITKCommon.dir/itkFloatingPointExceptions.cxx.o
> In file included from /private/var/folders/py/nyy7zwy505s32zc86rjh9vz80000gn/T/RtmpyN0ZpC/R.INSTALLc9b11febc519/ITKR/src/itks/Modules/Core/Common/src/itkFloatingPointExceptions.cxx:127:
> /private/var/folders/py/nyy7zwy505s32zc86rjh9vz80000gn/T/RtmpyN0ZpC/R.INSTALLc9b11febc519/ITKR/src/itks/Modules/Core/Common/src/itkFloatingPointExceptions_unix.cxx:140:3: error: use of undeclared identifier 'itk_feenableexcept'
>   itk_feenableexcept(FE_DIVBYZERO);
>   ^
> /private/var/folders/py/nyy7zwy505s32zc86rjh9vz80000gn/T/RtmpyN0ZpC/R.INSTALLc9b11febc519/ITKR/src/itks/Modules/Core/Common/src/itkFloatingPointExceptions_unix.cxx:141:3: error: use of undeclared identifier 'itk_feenableexcept'
>   itk_feenableexcept(FE_INVALID);
>   ^
> /private/var/folders/py/nyy7zwy505s32zc86rjh9vz80000gn/T/RtmpyN0ZpC/R.INSTALLc9b11febc519/ITKR/src/itks/Modules/Core/Common/src/itkFloatingPointExceptions_unix.cxx:162:3: error: use of undeclared identifier 'itk_fedisableexcept'
>   itk_fedisableexcept(FE_DIVBYZERO);
>   ^
> /private/var/folders/py/nyy7zwy505s32zc86rjh9vz80000gn/T/RtmpyN0ZpC/R.INSTALLc9b11febc519/ITKR/src/itks/Modules/Core/Common/src/itkFloatingPointExceptions_unix.cxx:163:3: error: use of undeclared identifier 'itk_fedisableexcept'
>   itk_fedisableexcept(FE_INVALID);
>   ^
> 4 errors generated.
> make[3]: *** [Modules/Core/Common/src/CMakeFiles/ITKCommon.dir/itkFloatingPointExceptions.cxx.o] Error 1
> make[2]: *** [Modules/Core/Common/src/CMakeFiles/ITKCommon.dir/all] Error 2
> make[2]: *** Waiting for unfinished jobs....
> [ 80%] Building CXX object Modules/ThirdParty/GDCM/src/gdcm/Source/MediaStorageAndFileFormat/CMakeFiles/gdcmMSFF.dir/gdcmPixmapReader.cxx.o

Thank you!

**Comment by stnava** [comment]
try latest version v0.6.0.0.1

**Comment by AlexandraBatzdorf** [comment]
Thanks for getting back to me! I tried installing the latest version, but it isn't working since cmaker needs to be version 3.16.3...3.19.7 or higher and I'm running version 3.11.4:
```
HEAD is now at 590dd29f00 ENH: Content link updates for ITK 5.3rc03
CMake Error at CMakeLists.txt:15 (cmake_minimum_required):
  CMake 3.16.3...3.19.7 or higher is required.  You are running version
  3.11.4
```
I tried reinstalling cmaker by running
`devtools::install_github("stnava/cmaker", force=T)`
but it reinstalled version 3.11.4. I also tried running
`devtools::install_github("stnava/ITKR", force=T, upgrade="always")`
but it gave me the same error message.
